datafusion/functions/src/macros.rs

```diff
@@ -122,6 +122,30 @@ macro_rules! make_package {
     };
 }
 
+/// Macro creates a sub module if the feature is not enabled
+///
+/// The rationale for providing stub functions is to help users to configure datafusion
+/// properly (so they get an error telling them why a function is not available)
+/// instead of getting a cryptic "no function found" message at runtime.
+
+macro_rules! make_stub_package {
```

Comment:
This is part of the `make_package` macro which I think keeps most of the "don't repeat yourself" goal but still allows `cargo fmt` to run.

Once I port over the other modules to use this pattern,  I will remove the `make_package` macro